### DISCUSSION TITLE
How to use ElasticSearch 7 with conductor 3.5.2

### DISCUSSION THREAD

**Discussion by shekarpcs** [Netflix/conductor#2826] [General]
I am trying to configure elasticSearch7 with conductor 3.5.2 with basic authentication username,password. 

if I do not set following property, Conductor Server able to connect to ES but it is not passing UID/PWD as ES6 implementation does not have way to pass uid/pwd.

conductor.elasticsearch.version=7

When I set above property, it is failing with classes not found. Could you guys provide proper details how to use ES7 and use ES7 implementation. I do not see any documentation or help

**Comment by erikbrgr** [comment]
Hi @shekarpcs, I think you're looking for this documentation?

https://github.com/Netflix/conductor/blob/b4673608bec51c3880ce69fc854a01c38fa0f13e/es7-persistence/README.md

I'm currently writing an article on setting up Conductor 3.5.1 with Elasticsearch 7.17.1 and PostgreSQL. If you find it useful, I'll happily post it here when it's finished? Hopefully, I'll complete it in the next few days.

**Comment by shekarpcs** [comment]
We are trying to use Basic authentication as our AWS es need it.

Es6 implementation does not have way to pass that.

Please post ur blog.

**Reply by erikbrgr** [reply]
I came across your previous post [here](https://github.com/Netflix/conductor/discussions/2789). Just wanted to let you know that using es7 does require you to build Conductor from source. There is no other way to do this, currently. My article is based on this, so it will not be of much help if you are using pre-compiled binaries.

That said, we use Docker to build, so maybe it'll still be of use considering your post [here](https://github.com/Netflix/conductor/discussions/2763)

I expect to publish my article in a few days

**Comment by lzslim** [comment]
Also stuck by es7 compiling error, waiting for blog...

**Comment by shekarpcs** [comment]
I tried to build based on documentation but build  fails.

So many unknowns.

**Comment by shekarpcs** [comment]
Could you please send me blog. I would like to still build it.

**Comment by erikbrgr** [comment]
Of course. I’ll put it here as soon as it is finished :)

On 13 Mar 2022, at 12:37, shekarpcs ***@***.***> wrote:

﻿

Could you please send me blog. I would like to still build it.

On Sat, Mar 12, 2022, 2:35 PM Shekar Chandu ***@***.***> wrote:

> I tried to build based on documentation but build fails.
>
> So many unknowns.
>
> On Sat, Mar 12, 2022, 11:40 AM Erik Burger ***@***.***>
> wrote:
>
>> I came across your previous post here
>> <https://github.com/Netflix/conductor/discussions/2789>. Just wanted to
>> let you know that using es7 does require you to build Conductor from
>> source. There is no other way to do this, currently. My article is based on
>> this, so it will not be of much help if you are using pre-compiled binaries.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Netflix/conductor/discussions/2826#discussioncomment-2347295>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AXQ2C7U2FPXCEGOV7J5RF5TU7TCIVANCNFSM5QE3BSQA>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>>
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

—
Reply to this email directly, view it on GitHub<https://github.com/Netflix/conductor/discussions/2826#discussioncomment-2349633>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AESRNUG3ALEUDM7U5MJDVGTU7XHRDANCNFSM5QE3BSQA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by shekarpcs** [comment]
What is the version you are building?
I am getting Jackson package not found in 3.5.x after all changes suggested.

**Comment by erikbrgr** [comment]
I’m building 3.5.1 without errors.

On 13 Mar 2022, at 12:49, shekarpcs ***@***.***> wrote:

﻿

What is the version you are building?
I am getting Jackson package not found in 3.5.x after all changes suggested.

On Sun, Mar 13, 2022, 7:47 AM Erik Burger ***@***.***> wrote:

> Of course. I’ll put it here as soon as it is finished :)
>
> On 13 Mar 2022, at 12:37, shekarpcs ***@***.***> wrote:
>
> ﻿
>
> Could you please send me blog. I would like to still build it.
>
> On Sat, Mar 12, 2022, 2:35 PM Shekar Chandu ***@***.***> wrote:
>
> > I tried to build based on documentation but build fails.
> >
> > So many unknowns.
> >
> > On Sat, Mar 12, 2022, 11:40 AM Erik Burger ***@***.***>
> > wrote:
> >
> >> I came across your previous post here
> >> <https://github.com/Netflix/conductor/discussions/2789>. Just wanted
> to
> >> let you know that using es7 does require you to build Conductor from
> >> source. There is no other way to do this, currently. My article is
> based on
> >> this, so it will not be of much help if you are using pre-compiled
> binaries.
> >>
> >> —
> >> Reply to this email directly, view it on GitHub
> >> <
> https://github.com/Netflix/conductor/discussions/2826#discussioncomment-2347295>,
>
> >> or unsubscribe
> >> <
> https://github.com/notifications/unsubscribe-auth/AXQ2C7U2FPXCEGOV7J5RF5TU7TCIVANCNFSM5QE3BSQA>
>
> >> .
> >> Triage notifications on the go with GitHub Mobile for iOS
> >> <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>
> >> or Android
> >> <
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> >>
> >> You are receiving this because you were mentioned.Message ID:
> >> ***@***.***>
> >>
> >
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/Netflix/conductor/discussions/2826#discussioncomment-2349633>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/AESRNUG3ALEUDM7U5MJDVGTU7XHRDANCNFSM5QE3BSQA>.
>
> Triage notifications on the go with GitHub Mobile for iOS<
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android<
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Netflix/conductor/discussions/2826#discussioncomment-2349651>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXQ2C7WCBUQ47VLSXCPNPYDU7XIVDANCNFSM5QE3BSQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

—
Reply to this email directly, view it on GitHub<https://github.com/Netflix/conductor/discussions/2826#discussioncomment-2349654>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AESRNUG7MGCFQCS3N7C23SLU7XI4FANCNFSM5QE3BSQA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by erikbrgr** [comment]
Hi @shekarpcs @lzslim , please see my blog post here: https://betterprogramming.pub/running-netflix-conductor-3-in-docker-using-elasticsearch-7-and-postgresql-b415988dd74a

Hope it helps!

**Comment by shekarpcs** [comment]
Ok. Thanks.

Are you following exactly same instructions as below?

Conductor 3.5.1
Jdk 11

This is what I am trying. First build successful. After changing stuff and
based on following and deleting dependency lock files. Boom. It failes.

https://github.com/Netflix/conductor/blob/b4673608bec51c3880ce69fc854a01c38fa0f13e/es7-persistence/README.md

**Comment by erikbrgr** [comment]
I was using the instructions as per my article on Medium here: https://betterprogramming.pub/running-netflix-conductor-3-in-docker-using-elasticsearch-7-and-postgresql-b415988dd74a

The difference between 3.5.1 and 3.5.2 should be negligible.

On 11 Oct 2022, at 09:49, shekarpcs ***@***.***> wrote:

﻿

Ok. Thanks.

Are you following exactly same instructions as below?

Conductor 3.5.1
Jdk 11

This is what I am trying. First build successful. After changing stuff and
based on following and deleting dependency lock files. Boom. It failes.

https://github.com/Netflix/conductor/blob/b4673608bec51c3880ce69fc854a01c38fa0f13e/es7-persistence/README.md

**Reply by ghost** [reply]
@erikbrgr did you known a "correct" way to implement API authentication for the conductor server ?

**Reply by erikbrgr** [reply]
@RubenMartins97 I'm afraid not, I haven't implemented authentication for the API. The Ops guys allowed access via an SSH key, but honestly that stuff is like magic to me :)

**Reply by ghost** [reply]
@erikbrgr Thanks for your response, maybe I will deal with this issue with an API gateway :).

**Comment by shekarpcs** [comment]
Es7 is only way. Es7 does not have in conductor

**Reply by ghost** [reply]
@shekarpcs  Did you handle this using es7?

**Comment by shekarpcs** [comment]
We tested with es 7 , it  worked . We compiled with es7. Not yet in
production.

**Reply by ghost** [reply]
@shekarpcs  Right I will test this solution, thanks for the feedback  :)
